pkg/cluster/volumes.go

```diff
@@ -0,0 +1,178 @@
+package cluster
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/client-go/pkg/api/resource"
+	"k8s.io/client-go/pkg/api/v1"
+
+	"github.com/zalando-incubator/postgres-operator/pkg/spec"
+	"github.com/zalando-incubator/postgres-operator/pkg/util"
+	"github.com/zalando-incubator/postgres-operator/pkg/util/constants"
+	"github.com/zalando-incubator/postgres-operator/pkg/util/filesystems"
+	"github.com/zalando-incubator/postgres-operator/pkg/util/volumes"
+)
+
+func (c *Cluster) listPersistentVolumeClaims() ([]v1.PersistentVolumeClaim, error) {
+	ns := c.Metadata.Namespace
+	listOptions := v1.ListOptions{
+		LabelSelector: c.labelsSet().String(),
+	}
+
+	pvcs, err := c.KubeClient.PersistentVolumeClaims(ns).List(listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("could not list of PersistentVolumeClaims: %v", err)
+	}
+	return pvcs.Items, nil
+}
+
+func (c *Cluster) deletePersistenVolumeClaims() error {
+	c.logger.Debugln("Deleting PVCs")
+	ns := c.Metadata.Namespace
+	pvcs, err := c.listPersistentVolumeClaims()
+	if err != nil {
+		return err
+	}
+	for _, pvc := range pvcs {
+		c.logger.Debugf("Deleting PVC '%s'", util.NameFromMeta(pvc.ObjectMeta))
+		if err := c.KubeClient.PersistentVolumeClaims(ns).Delete(pvc.Name, c.deleteOptions); err != nil {
```

Comment:
I think it would be wise to get the namespace from ``pvc``, not from the cluster:
``c.KubeClient.PersistentVolumeClaims(pvc.Namespace).``